### DISCUSSION TITLE
ING Germany change description

### DISCUSSION THREAD

**Discussion by M0nk3yOo** [firefly-iii/firefly-iii#8419] [Firefly III] [answered]
Hi everybody,

I could find any useful answer here :-(

Does anybody found any solution to change the description to a more cleaner one for ING German accounts, imported via GoCardless

<img width="1124" alt="image" src="https://github.com/firefly-iii/firefly-iii/assets/32765291/b38af65a-1038-4566-959f-d581bfd9f747">

A "replace-rule" could be useful, but seems not existing.

Thanks a lot!

**Comment by JC5** [comment]
This is something that Nordigen  /GoCardless support can pick up.

**Reply by M0nk3yOo** [reply]
So, they'll know the firefly solution if I mention it?

**Reply by JC5** [reply]
Yea they should!

**Reply by M0nk3yOo** [reply]
Okay, thanks. I opened a ticket. Let's wait for any feedback.

**Reply by JC5** [reply]
Wait, windows. Try "c:\\\bla"

**Reply by M0nk3yOo** [reply]
hm? I don't understand. Wrong thread?

(I'm using firefly via Docker)

**Reply by JC5** [reply]
Whoops yes, my bad!

**Reply by M0nk3yOo** [reply]
The support is now asking me for those information:
account_id, requisition_id (connection) 

Do you have a hint where I can find those information?

**Reply by JC5** [reply]
You can find them in the JSON config file, or when you reconnect the data importer, you can first download the configuration file and grab them.

**Reply by M0nk3yOo** [reply]
Thanks, I was thinking about that but was not sure:
<img width="627" alt="image" src="https://github.com/firefly-iii/firefly-iii/assets/32765291/4ab3f6c5-74f0-48f7-8b2e-2fecb4dd2814">

So the first pixel out account is the account_id, and at the bottom pixel out is the requisition_id (left side/first part of the value)

**Reply by M0nk3yOo** [reply]
Hmmm... here is the reply. Seems (I understand it like that) that they can't do anything.
So, maybe a "replace" rule might be good :-/



> I reviewed the original response, and the value dates that the bank returns align with what we transmit via the API. They include the remittanceInformationUnstructured, which we pass on to you. However, we don't receive this field from them as a description (Beschreibung), causing differences in information. The bank includes various details under this field based on their criteria. It's essential to note that data formatting and information may vary between the bank statement and the PSD2 API.
>  
> More information about what kind of information is available and the formatting can be found in banks developer documentation. 
>  
> Let me know if additional questions remain.

**Reply by JC5** [reply]
Sorry, [there will not be a "replace" rule](https://docs.firefly-iii.org/explanation/more-information/what-its-not/#more-complicated-rules-like-boolean-logic-copypasting-values-etc). There are two solutions.

1. Use a webhook to clean up transactions.
2. Contact your bank, and see if they wish to update these transactions on their end.

**Reply by M0nk3yOo** [reply]
I wrote a little script to solve my issue. Not that pretty, but it works for me.
Unfortunately I can't upload it here (it's a js). This editor here destroy the structure and probably build in some char errors.





const http = require('http');

let hostname = '<hostname/ip>';
let port = <port>;
let token =
	'<personal token>';

const options = {
	hostname: hostname,
	port: port,
	path: '/api/v1/transactions?limit=60',
	// path: '/api/v1/transactions?limit=10&page=1&start=2024-01-10&end=2024-01-31',
	method: 'GET',
	headers: {
		'Content-Type': 'application/json',
		accept: 'application/vnd.api+json',
		Authorization: `Bearer ${token}`,
	},
};

const getPosts = () => {
	let data = '';

	const request = http.request(options, (response) => {
		response.setEncoding('utf8');
		response.on('data', (chunk) => {
			data += chunk;
		});

		response.on('end', () => {
			transactions = JSON.parse(data).data;

			transactions.map(async (transaction) => {
				let id = transaction.id;
				let description = transaction.attributes.transactions[0].description;

				if (description.includes('remittanceinformation')) {
					// console.log(id, 'Muss geändert werden');
					description = description.split(',');

					description.map((entry) => {
						if (entry.includes('remittanceinformation:')) {
							transaction.attributes.transactions[0].description = entry.replace(/(remittanceinformation:)/g, '');
							let transaction_new = transaction.attributes.transactions;

							//  Send PUT to backend
							const options2 = {
								hostname: hostname,
								port: port,
								path: `/api/v1/transactions/${id}`,
								method: 'PUT',
								headers: {
									'Content-Type': 'application/json',
									accept: 'application/vnd.api+json',
									Authorization: `Bearer ${token}`,
								},
							};

							// console.log(options2);
							// console.log(transaction_new);

							let data2 = '';

							const request2 = http.request(options2, (response) => {
								response.setEncoding('utf8');
								response.on('data', (chunk2) => {
									data2 += chunk2;
								});

								response.on('end', () => {
									console.log(data2);
								});
							});

							request2.on('error', (error) => {
								console.error(error);
							});

							request2.write(
								JSON.stringify({
									// apply_rules: false,
									// fire_webhooks: true,
									// group_title: 'Split transaction title.',
									transactions: transaction_new,
								})
							);

							request2.end();
						}
					});
				}
			});
		});
	});

	request.on('error', (error) => {
		console.error(error);
	});

	request.end();
};

getPosts();

**Reply by JC5** [reply]
That's pretty cool! But yep hard to read lol. 

Have you tried https://gist.github.com ?

**Reply by M0nk3yOo** [reply, marked as answer]
Here you are: https://gist.github.com/M0nk3yOo/5021ff346301de99cb1dc7fe6aeec1c6

**Reply by philnse** [reply]
First of all, thanks for this script, thats exactly what I was looking for! What would be the easiest way to run this? @M0nk3yOo

**Reply by M0nk3yOo** [reply]
Yes, sure! :-)

I've configured an cronjob, which runs the following:

`node /volume1/docker/firefly/firefly_modify_transactions.js`

Node must be installed on the device, where the cron runs.
The script should then contact the firefly instance and doing the corrections (could be, that you need to do some addtional modification/adds on the script)


P.S.: I'm currently not using firefly because of the overhead :-)
